src/components/Content.jsx

```diff
@@ -6,16 +6,16 @@ function Content() {
     
     const {
         recipes,
-        offset,
-        setOffset,
+        urlEndpoints,
+        setUrlEndpoints,
         noOfResults,
     } = useContext(RecipeContext)
 
     const handleOffset = (e)=>{
-        const currentOffset = offset
-        e.target.innerText === 'Previous' && currentOffset >= noOfResults ? setOffset(currentOffset - noOfResults) :
-        e.target.innerText === 'Next' ? setOffset(currentOffset + noOfResults) : 
-        offset === 0 ? setOffset(0) : setOffset(0)
+        const currentOffset = urlEndpoints.offset
+        e.target.innerText === 'Previous' && currentOffset >= noOfResults ? setUrlEndpoints({...urlEndpoints,offset:(currentOffset - noOfResults)}) :
+        e.target.innerText === 'Next' ? setUrlEndpoints({...urlEndpoints,offset:(currentOffset + noOfResults)}) : 
```

Comment:
Instead of passing the rendered DOM Element `e` and then referencing it `e.target.innerText` to determine what the user clicked. Use the `onClick` eventListener you have, `handleOffset()` and instead of passing `e` pass a string `prev` or `next`. That's a suggestion that cleans this up a little.

Also, this is nit picky, but these complete ternary conditions are difficult to read. You'll see on big teams, there is a preference to extract whatever you can to make this easy to ready as you go along. If you need to nest a terinary, break it down. IE:

``` if you update to my earlier suggestion
if(location === 'Prev' && currentOffset >= noOfResults) 
  setUrlEndpoints({...urlEndpoints,offset:(currentOffset - noOfResults)})

if(location === 'Next')
  setUrlEndpoints({...urlEndpoints,offset:(currentOffset + noOfResults)})    

if(location === 0)
  setUrlEndpoints({...urlEndpoints,offset:0})
```
const handleOffset = (location) => {
  ... // other code you have
  if(l

Reply:
thanks for the tip on the ternary expressions, I find them easy to read when written like that but its good to know that its not preferred practice.

will update per  your suggestion on the prev and next as well.

Reply:
To be more clear, ternary's are fine to write. It's when they get so long they start wrapping or when you have to mentally paint a picture that starts to get complex.

Reply:
thanks for the clarification. I understand. I have that changed to

`const handleOffset = (location)=>{
        const currentOffset = urlEndpoints.offset
        if(location === 'previous' && currentOffset >= urlEndpoints.noOfResults)
            setUrlEndpoints({...urlEndpoints,offset:(currentOffset - urlEndpoints.noOfResults)})
        if(location === 'next')
            setUrlEndpoints({...urlEndpoints,offset:(currentOffset + urlEndpoints.noOfResults)})
        if(urlEndpoints.offset === 0 && location === 'previous')
            setUrlEndpoints({...urlEndpoints,offset:0})
    }`

    I will start using ternary's for simple statements and using `if` for complex conditions.
    also I can change `location`  to `navOffset`